src/PkgAuthentication.jl

```diff
@@ -175,8 +175,31 @@ end
 Base.show(io::IO, s::NoAuthentication) = print(io, "NoAuthentication($(s.server))")
 
 function step(state::NoAuthentication)::Union{RequestLogin, Failure}
-    challenge = Random.randstring(32)
     output = IOBuffer()
+    # Try device flow first
+    response = Downloads.request(
+        joinpath(state.server, "dex/device/code"),
```

Comment:
We should discover this using the "well-known openid configuration" URL instead of hard coding it. It will also indicate if the authsrvr supports device login at all.

Reply:
Done in https://github.com/JuliaComputing/PkgAuthentication.jl/pull/42/commits/79e417d17ba77db3945a743853aa08d7a126385e